sktime/dists_kernels/tests/test_all_dist_kernels.py

```diff
@@ -0,0 +1,135 @@
+from sktime.utils import all_estimators
+from sktime.utils._testing.panel import make_transformer_problem
+from sktime.tests._config import ESTIMATOR_TEST_PARAMS
+
+PAIRWISE_TRANSFORMERS_TAB = all_estimators(
+    estimator_types="transformer-pairwise-tabular", return_names=False
+)
+PAIRWISE_TRANSFORMERS_PAN = all_estimators(
+    estimator_types="transformer-pairwise-panel", return_names=False
+)
+
+X1_tab = make_transformer_problem(
+    n_instances=4,
+    n_columns=4,
+    n_timepoints=5,
+    random_state=1,
+    return_numpy=True,
+    panel=False,
+)
+X2_tab = make_transformer_problem(
+    n_instances=5,
+    n_columns=5,
+    n_timepoints=5,
+    random_state=2,
+    return_numpy=True,
+    panel=False,
+)
+
+X1_tab_df = make_transformer_problem(
+    n_instances=4,
+    n_columns=4,
+    n_timepoints=5,
+    random_state=1,
+    return_numpy=False,
+    panel=False,
+)
+X2_tab_df = make_transformer_problem(
+    n_instances=5,
+    n_columns=5,
+    n_timepoints=5,
+    random_state=2,
+    return_numpy=True,
+    panel=False,
+)
+
+
+@pytest.mark.parametrize("pairwise_transformers_tab", PAIRWISE_TRANSFORMERS_TAB)
+def test_pairwise_transformers_tab(pairwise_transformers_tab):
+    # Test numpy tabular
+    _general_pairwise_transformer_tests(X1_tab, X2_tab, pairwise_transformers_tab)
+
+    # Test dataframe tabular
+    _general_pairwise_transformer_tests(X1_tab_df, X2_tab_df, pairwise_transformers_tab)
+
+
+X1_num_pan = make_transformer_problem(
+    n_instances=4, n_columns=4, n_timepoints=5, random_state=1, return_numpy=True
+)
+X2_num_pan = make_transformer_problem(
+    n_instances=5, n_columns=5, n_timepoints=5, random_state=2, return_numpy=True
+)
+
+X1_list_df = make_transformer_problem(
+    n_instances=4, n_columns=4, n_timepoints=5, random_state=1, return_numpy=False
+)
+X2_list_df = make_transformer_problem(
+    n_instances=5, n_columns=5, n_timepoints=5, random_state=2, return_numpy=False
+)
+
+X1_num_df = np.array(X1_list_df)
+X2_num_df = np.array(X2_list_df)
+
+
+@pytest.mark.parametrize("pairwise_transformers_pan", PAIRWISE_TRANSFORMERS_PAN)
+def test_pairwise_transformers_pan(pairwise_transformers_pan):
+    test_params = ESTIMATOR_TEST_PARAMS[pairwise_transformers_pan]
+
+    # Test numpy panel (3d numpy)
+    _general_pairwise_transformer_tests(
+        X1_num_pan, X2_num_pan, pairwise_transformers_pan, test_params
+    )
+
+    # Test list of dataframes
+    _general_pairwise_transformer_tests(
+        X1_list_df, X2_list_df, pairwise_transformers_pan, test_params
+    )
+
+    # Test numpy of dataframes
+    _general_pairwise_transformer_tests(
+        X1_num_df, X2_num_df, pairwise_transformers_pan, test_params
+    )
+
+
+def _general_pairwise_transformer_tests(x, y, pairwise_transformers_tab, kwargs=None):
+    if kwargs is not None:
+        transformer = pairwise_transformers_tab(**kwargs)
+    else:
+        transformer: BasePairwiseTransformer = pairwise_transformers_tab()
```

Comment:
Minor question:
Does adding the type hint change how the test runs, or is it more for readability?

Reply:
Sorry thats just me being inconsistent I normally type a lot of my Python but try to avoid typing things in tests since you can get some bad interaction between it and Pytest and so this shouldn't be typed.